File: src/main/java/models/Mission.java
package models;

public class Mission {
    int id;
    String description;
    int points_recompense;
    String statut;

    public Mission() {};

    public Mission(int id, String description, int points_recompense, String statut) {
        this.id = id;
        this.description = description;
        this.points_recompense = points_recompense;
        this.statut = statut;
    }
    public Mission(String description, int points_recompense, String statut) {
        this.id = id;
        this.description = description;
        this.points_recompense = points_recompense;
        this.statut = statut;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getPoints_recompense() {
        return points_recompense;
    }

    public void setPoints_recompense(int points_recompense) {
        this.points_recompense = points_recompense;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    @Override
    public String toString() {
        return "Mission{" +
                "id=" + id +
                ", description='" + description + '\'' +
                ", points_recompense=" + points_recompense +
                ", statut='" + statut + '\'' +
                '}';
    }
}

File: ProjetWebTrekSwap/src/Entity/Mission.php
<?php

namespace App\Entity;

use App\Repository\MissionRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: MissionRepository::class)]
class Mission
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(nullable: true)]
    #[Assert\NotNull(message: "L'ID de la recompense ne peut pas être nul.")]
    private ?int $id_recompense = 0;


    #[ORM\Column(length: 255)]
    private ?string $description = null;

    #[ORM\Column]
    private ?int $points_recompense = null;

    #[ORM\Column(length: 255)]
    private ?string $statut = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function getPointsRecompense(): ?int
    {
        return $this->points_recompense;
    }

    public function setPointsRecompense(int $points_recompense): static
    {
        $this->points_recompense = $points_recompense;

        return $this;
    }

    public function getStatut(): ?string
    {
        return $this->statut;
    }

    public function setStatut(string $statut): static
    {
        $this->statut = $statut;

        return $this;
    }
}
